probe-rs/src/flashing/download.rs

```diff
@@ -145,6 +145,31 @@ pub fn download_file_with_options<P: AsRef<Path>>(
         .map_err(FileDownloadError::Flash)
 }
 
+/// Mass-erase all nonvolatile memory.
+pub fn erase_all(session: &mut Session) -> Result<(), FlashError> {
+    log::debug!("Erasing all...");
+    let algos = session.target().flash_algorithms.clone();
+    for algorithm in algos {
+        log::debug!("Erasing with algorithm: {}", algorithm.name);
+
+        let mut flasher = Flasher::new(session, &algorithm)?;
+
+        // If the flash algo doesn't support erase all, disable chip erase.
+        if !flasher.is_chip_erase_supported() {
+            // TODO: loop over all sectors erasing them individually instead.
+            log::warn!(
+                "Chip erase not supported for algorithm {}, skipping.",
+                algorithm.name
+            );
```

Comment:
I think we should return an error if flashing all regions is not possible: Right now it could happen that nothing gets erased, but the caller is not informed.

Reply:
True, that's a concern. What should the exact behavior be? Return error if some algo doesn't support chip erase? or only if no algo supports chip erase?

Some will never support chip erase, such as STM32's OTPs, so the first one will be somewhat useless.

Reply:
On the other hand the second is still "dangerous" as in no error will be returned if eg option bytes are mass-erased but not the main flash, which is definitely not what the user expects...

Reply:
We could also manually (i.e. sector by sector) erase the regions where chip erase is is not possible?